packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -61,6 +62,10 @@ class FlutterOptions {
 }
 
 abstract class FlutterCommand extends Command<Null> {
+  @override
+  ArgParser get argParser => _argParser;
+  final ArgParser _argParser = new ArgParser(allowTrailingOptions: false);
```

Comment:
The default value of allowTrailingOptions has flipped.  This restores the old behavior.